### DISCUSSION TITLE
How to use writeFragment?

### DISCUSSION THREAD

**Discussion by cevr** [urql-graphql/urql#2098] [Q&A] [answered]
Im using writeFragment to update a row in a list I'm displaying. The fragment reads correctly, but it doesn't seem to be propagating the update when I try to write.

https://gist.github.com/cevr/9ab1fe498ba8383d13fab8c7aaca26c8

The event is never updated. Am I missing something?

**Comment by JoviDeCroock** [comment, marked as answer]
Have you tried adding `cases` to your fragment, else it won't be able to write 😅 it uses the fields in your fragment to know what you want it to write.

So:

```js
  cache.writeFragment(
    gql`
      fragment _ on CaseManagementEvent {
        id
        cases { id status __typename }
      }
    `,
    {
      id: variables.eventId,
      cases: [{ ...result.createCaseFromEvent, __typeneame: 'Case' }]
    },
  );
```

**Reply by cevr** [reply]
Oh really? https://formidable.com/open-source/urql/docs/api/graphcache/#writefragment says that I didnt have to haha

![image](https://user-images.githubusercontent.com/34170625/143296154-4451a613-3069-4c08-97ba-6a05c672eda9.png)

**Reply by JoviDeCroock** [reply]
Hmm, let me double check, I think this is just for the key-creation but if so that should be clarified

**Reply by cevr** [reply]
Adding the __typename to the fragment ended up working 😅  but it didnt propagate the change to the row

**Reply by JoviDeCroock** [reply]
If you have a reproduction I'd be open to having a look, checked our testing suite and everything seems ok. Example https://github.com/FormidableLabs/urql/blob/main/exchanges/graphcache/src/store/store.test.ts#L508

**Reply by cevr** [reply]
yeah will try to make one!
